packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -176,10 +176,8 @@ export class ApolloServerBase {
       ...requestOptions
     } = config;
 
-    if (
-      gateway &&
-      (modules || schema || typeDefs || subscriptions || resolvers)
-    ) {
+    if (gateway && (modules || schema || typeDefs || resolvers)) {
+      // TODO: this could be handled by adjusting the typings to keep gateway configs and non-gateway configs seprate.
```

Comment:
While true, keep in mind that Apollo Server is very frequently consumed without TypeScript, so this is still important!